sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java

```diff
@@ -190,6 +193,32 @@ public DirectConnectionConfig setMaxRequestsPerConnection(int maxRequestsPerConn
         return this;
     }
 
+    /**
+     * Gets the request timeout interval
+     * This represents the timeout interval for requests
+     *
+     * Default value is 60 seconds
+     *
+     * @return the request timeout interval
+     */
+    public Duration getRequestTimeout() {
```

Comment:
Is this the end to end (including the consistency stack) timeout?

Reply:
yes.

Reply:
@kirankumarkolli 
both GWConnectionConfig and DirectConnectionConfig have requestTimeout.
and `DirectConnectionConfig#requestTimeout` represents the end to end latency (including consistency stack).

I wonder if we should keep these at the ConnectionConfig level or if we should move it up to the CosmosClientBuilder. How .Net is doing this?

Reply:
@moderakh  - Let's merge this PR as it is, and we will have a follow up PR which will handle these changes - https://github.com/Azure/azure-sdk-for-java/issues/11531

Reply:
agreed. closing this as resolved.